packages/dds/tree/src/simple-tree/api/discrepancies.ts

```diff
@@ -127,30 +97,35 @@ export interface AllowedTypeDiscrepancy extends FieldDiscrepancyLocation {
 	readonly stored: readonly TreeNodeSchemaIdentifier[];
 }
 
+/**
+ * Differences in `FieldKindIdentifier` between two schema.
+ */
 export interface FieldKindDiscrepancy extends FieldDiscrepancyLocation {
 	readonly mismatch: "fieldKind";
 	readonly view: FieldKindIdentifier;
 	readonly stored: FieldKindIdentifier;
 }
 
+/**
+ * Differences in the `ValueSchema` of two `LeafNodeStoredSchema` objects.
+ */
 export interface ValueSchemaDiscrepancy {
 	identifier: TreeNodeSchemaIdentifier;
 	mismatch: "valueSchema";
 	view: ValueSchema | undefined;
 	stored: ValueSchema | undefined;
 }
 
+/**
+ * Differences in the kind of node schema.
+ *
+ * Includes when stored object schema are expected to be compatible with an array node schema.
+ */
 export interface NodeKindDiscrepancy {
 	identifier: TreeNodeSchemaIdentifier;
 	mismatch: "nodeKind";
-	view: SchemaFactoryNodeKind | undefined;
-	stored: SchemaFactoryNodeKind | undefined;
-}
-
-export interface NodeFieldsDiscrepancy {
```

Comment:
I just inclined the differences into the top level list of discrepancies since they contain the node identifier already I don't think anything of value is lost from that.